app/controllers/registration_controller.rb

```diff
@@ -5,9 +5,10 @@ def index
 
   def create
     @user = User.new(user_params)
-    if @user.save
-      redirect_to home_index_path
-    end
+    return render 'index' unless @user.save
+    credential = Credential.new(email: @user.email, password: @user.password)
```

Comment:
what's the point with credential if has_secure_password is doing the job?
also if the user is new you don't need to validate the password with authenticate

Reply:
Ok, let me fix all those errors, indeed not need of validation